python/setup.py

```diff
@@ -139,7 +139,7 @@
     "grpcio >= 1.28.1",
     "jsonschema",
     "msgpack >= 1.0.0, < 2.0.0",
-    "numpy >= 1.16",
+    "numpy >= 1.19.3",
```

Comment:
```suggestion
    "numpy >= 1.16; python_version < '3.9'",
    "numpy >= 1.19.3; python_version >= '3.9'",
```

Reply:
> Going from 1.16 to 1.19 everywhere is a pretty aggressive bump in the numpy-requirements.

Why do you say this?
The tests pass just fine with numpy 1.19.3 and it has built wheels back to py36.

Reply:
Due to changing ABIs between versions, the numpy-version at install-time becomes a hard lower bound for runtime as well (essentially, it can happen that the wheel links against newer symbols that might not be present on older numpy versions, leading to possible runtime breakage). For this reason, conda-forge (and other projects) generally try to _build_ with the oldest reasonable numpy version.

I know that it's necessary for python 3.9, but bumping the requirement uniformly across all python versions means that users will not be able anymore to install ray without having the newest numpy. Hard-requiring such a recent version can make various upgrade-paths much more narrow and complicated.

So this has nothing to do with the tests. In any case, there was no criticism implied, only an outside observation that this seems more eager than what's necessary.

Reply:
> Hard-requiring such a recent version can make various upgrade-paths much more narrow and complicated.

that is def true, let's see what @wuisawesome says and then I'll add that in.

Reply:
Yup, I agree with this suggestion/rationale. I don't think we need to worry about these cleanup details until we get a working wheel though.